endpoint.bal

```diff
@@ -0,0 +1,372 @@
+        string pathWithReadMasks = prepareUrlWithReadMasks(path, readMasks);
+        string pathWithQuery = pathWithReadMasks + QUERY_PATH + query;
+        http:Response httpResponse = <http:Response>check self.googleContactClient->get(pathWithQuery);
+        var response = check handleResponseWithNull(httpResponse);
+        log:printInfo(response.toString());
+        var searchResponse = check response.cloneWithType(SearchResponse);
+        log:printInfo(searchResponse.toString());
+        PersonResponse[] persons = [];
+        int i = persons.length();
+        foreach var result in searchResponse.results {
+            var personResult = result.person;
+            if(personResult is json) {
+                persons[i] = check personResult.cloneWithType(PersonResponse);
+                i = i + 1;
+            }
+        }
+        return persons;
+    }
+
+    # Update contact photo for a contact.
+    # 
+    # + resourceName - Contact resource name
+    # + imagePath - Path to image from root directory
+    # + return - () on success, else an 'error'
+    @display {label: "Update a Contact Photo"}
+    isolated remote function updateContactPhoto(@display {label: "Contact Resource Name"} string resourceName,
+                                       @display {label: "Image Path"} string imagePath) returns 
+                                       @tainted error? {
+        string path = SLASH + resourceName + COLON + UPDATE_PHOTO_PATH;
+        http:Request request = new;
+        string encodedString = check convertImageToBase64String(imagePath);
+        json updatePayload = {"photoBytes": encodedString};
+        request.setJsonPayload(updatePayload);
+        http:Response uploadResponse = <http:Response>check self.googleContactClient->patch(path, request);
+        return handleUploadPhotoResponse(uploadResponse);
+    }
+
+    # Delete a contact photo.
+    # 
+    # + resourceName - Contact resource name
+    # + return - () on success, else an 'error'
+    @display {label: "Delete a Contact Photo"}
+    isolated remote function deleteContactPhoto(@display {label: "Contact Resource Name"} string resourceName) returns 
+                                       @tainted error? {
+        string path = SLASH + resourceName + COLON + DELETE_PHOTO_PATH;
+        http:Response deleteResponse = <http:Response>check self.googleContactClient->delete(path);
+        return handleDeleteResponse(deleteResponse);
+    }
+
+    # Batch get contacts.
+    # 
+    # + resourceNames - String array of contact resource names
+    # + personFields - Restrict which fields on the person are returned
+    # + return - `PersonResponse[]` on success, else an `error`
+    @display {label: "Batch Get Contacts"}   
+    isolated remote function batchGetContacts(@display {label: "Contact Resource Names"} string[] resourceNames, 
+                                     @display {label: "Person Fields"} ContactMasks[] personFields) returns 
```

Comment:
`Person Fields` will this be meaningful for a user? What he needs to fill here?

Reply:
Person fields denote what are fields to return